### ISSUE TITLE
Add support for fontFeatureSettings in theme.json

### ISSUE THREAD

**Issue by bgardner** [WordPress/gutenberg#56619]
## What problem does this address?
With all of the fonts available to theme developers, it would be great to add support for fontFeatureSettings in theme.json. AFAIK, there is no way to add this setting on a per-font basis. Putting this in a style sheet is not ideal for themes that provide multiple fonts, as there could be some undesirable outcomes depending on which font is selected.

## What is your proposed solution?
Include support for font-feature-settings similar to this:

```
"fontFamily": "Inter, sans-serif",
"name": "Inter",
"slug": "inter",
"fontFace": [
	{
		"fontDisplay": "block",
		"fontFamily": "Inter",
		"fontFeatureSettings": 'cv06', 'cv10', 'cv11', 'ss01', 'ss07', 'ss08',
		"fontStretch": "normal",
		"fontStyle": "normal",
		"fontWeight": "100 900",
		"src": [ "file:./assets/fonts/Inter-Variable.woff2" ]
	}
]
```

**Comment by bgardner** [comment]
Noting that this appears to exist (https://github.com/WordPress/gutenberg/discussions/47127) but is not working as expected.

**Comment by bgardner** [comment]
@jordesign @richtabor Can one of you switch the tags on this from Enhancement to Bug?

And upon further review, the @font-face rules seem to be in place, but not working:

```
<style id='wp-fonts-local'>
@font-face{font-family:Inter;font-style:normal;font-weight:100 900;font-display:block;src:url('http://localhost:10009/wp-content/themes/powder/assets/fonts/Inter-Variable.woff2') format('woff2');font-feature-settings:'cv06', 'cv10', 'cv11', 'ss01', 'ss07', 'ss08';font-stretch:normal;}
</style>
```

However, if I add the following CSS in style sheet, it works:

```
body {
	font-feature-settings: 'cv06', 'cv10', 'ss01', 'ss07', 'ss08';
}
```

**Comment by jordesign** [comment]
Sure can @bgardner  :)

**Comment by t-hamano** [comment]
@bgardner Which browser are you testing with? According to [Can I use](https://caniuse.com/mdn-css_at-rules_font-face_font-feature-settings), `font-feature-settings` inside `@font-face` doesn't seem to be supported in Chrome or Edge.

**Comment by bgardner** [comment]
@t-hamano Well snap, that’s something I didn't expect. Usually Chrome is at the forefront of these things, but after testing, I believe you are right. (Which is kind of annoying because Chrome is such a large % of browser.)

**Comment by t-hamano** [comment]
I don't know why this feature got supported, but technically this issue is a browser support issue, not a Gutenberg issue. So while it's hard to call this a bug, we feel it might be a good idea to update the JSON Schema description to indicate that this feature is not yet supported by major browsers.

![font-feature-settings](https://github.com/WordPress/gutenberg/assets/54422211/8a7e5c24-449d-44a2-9199-55185b90dd1d)

**Comment by JanHoek** [comment]
I tested this in Chrome, Firefox, and Safari—on both desktop and mobile.  
It appears to work fine this way, and you can still override it in your regular stylesheet.

```json
{
  "fontFamily": "Mona Sans, sans-serif",
  "name": "Mona Sans",
  "slug": "mona-sans",
  "fontFace": [
    {
      "fontFamily": "Mona Sans",
      "src": [
        "file:./assets/fonts/monasans.woff2"
      ],
      "fontVariationSettings": "'wdth' 104, 'opsz' 6",
      "fontFeatureSettings": "'ss03' on",
      "fontWeight": "200 900",
      "fontStretch": "75% 125%",
      "fontDisplay": "swap"
    }
  ],
  "blocks": {
    "core/list": {
      "css": "font-variation-settings:\"wdth\" 102,\"opsz\" 6; font-feature-settings:\"ss03\" on;"
    }
  },
  "elements": {
    "h1": {
      "css": "font-variation-settings:\"wdth\" 95,\"opsz\" 11; font-feature-settings:\"ss05\" off;"
    }
  }
}

**Comment by bgardner** [comment]
Oh wow, it’s possible that Chrome supports this now—as I have just tested this myself and it seems to be working.

**Comment by t-hamano** [comment]
Yes, [font-feature-settings descriptor in @font-face rule is now supported in Chrome 139](https://developer.chrome.com/release-notes/139#support_font-feature-settings_descriptor_in_font-face_rule). I think this issue can be closed.
